File: src/Chapter-6-C99/Questions/Question8.c
//Write a program that prints a one-month calendar. The user specifies the number
//of days in the month and the day of the week on which the month begins
#include <stdio.h>

int main(void)
{
    int days, start;
    printf("Enter a number of days in month: ");
    scanf("%d", &days);
    printf("Enter starting day of the week (1=Sun, 7=Sat): ");
    scanf("%d", &start);

    for (int i=0; i<=start; i++)
        printf("  ");
    for (int i=1; i<=days; start++, i++)
    {
        printf("%4d", i);
        if (start%7 == 0)
        {
            printf("\n");
        }
    }
    return 0;
}

File: src/Chapter-6-Rust/Questions/Question8.rs
//Write a program that prints a one-month calendar. The user specifies the number
//of days in the month and the day of the week on which the month begins
#[macro_use] extern crate text_io;
use std::process::Command;
//Need this to use the print!
use std::io::{self, Write};

fn main()
{
	let (days, mut start) : (i32, i32);
	print!("Enter a number of days in month: ");
	io::stdout().flush().unwrap();
	scan!("{}", days);
	print!("Enter a starting day of the week (1=Sun, 7=Sat): ");
	io::stdout().flush().unwrap();
	scan!("\n{}", start);
	for _ in 0..start
	{
		print!("   ");
		io::stdout().flush().unwrap();
	}
	for i in 1..days
	{
		print!("{:4}", i);
		io::stdout().flush().unwrap();
		if start%7 == 0
		{
			print!("\n");
			io::stdout().flush().unwrap();
		}
		start += 1;
	}
	println!("");
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
